starter-code/pizza.js

```diff
@@ -1,85 +1,168 @@
 // Write your Pizza Builder JavaScript in this file.
 
-// Constants 
-var basePrice = 10
-var ingredients = {
-  pepperonni: {name: 'Pepperonni', price: 1},
-  mushrooms: {name: 'Mushrooms', price: 1},
-  greenPeppers: {name: 'Green Peppers', price: 1},
-  whiteSauce: {name: 'White sauce', price: 3},
-  glutenFreeCrust: {name: 'Gluten-free crust', price: 5}
-}
+// Constants
+let basePrice = 10;
+let ingredients = {
+  pepperonni: { name: 'Pepperonni', price: 1 },
+  mushrooms: { name: 'Mushrooms', price: 1 },
+  greenPeppers: { name: 'Green Peppers', price: 1 },
+  whiteSauce: { name: 'White sauce', price: 3 },
+  glutenFreeCrust: { name: 'Gluten-free crust', price: 5 },
+};
 
 // Initial value of the state (the state values can change over time)
-var state = {
+let state = {
   pepperonni: true,
   mushrooms: true,
   greenPeppers: true,
   whiteSauce: false,
-  glutenFreeCrust: false
-}
+  glutenFreeCrust: false,
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll('.pep').forEach(function($pep) {
     if (state.pepperonni) {
-      $pep.style.visibility = "visible";
-    }
-    else {
-      $pep.style.visibility = "hidden";
+      $pep.style.visibility = 'visible';
+    } else {
+      $pep.style.visibility = 'hidden';
     }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mushrooms) {
+    if (state.mushrooms) {
+      $mushrooms.style.visibility = 'visible';
+    } else {
+      $mushrooms.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function($greenPeppers) {
+    if (state.greenPeppers) {
+      $greenPeppers.style.visibility = 'visible';
+    } else {
+      $greenPeppers.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function($sauce) {
+    if (state.sauce) {
+      $sauce.style.visibility = 'visible';
```

Comment:
In this case you should have used the .add and .remove methods.
In this case, in my local tests the white sauce does not render.